main/theme/createCtx.tsx

```diff
@@ -2,21 +2,15 @@ import React from 'react';
 
 type CreateCtx<A> = readonly [
   () => A,
-  React.ProviderExoticComponent<React.ProviderProps<A | undefined>>,
+  React.ProviderExoticComponent<React.ProviderProps<A>>,
 ];
 
 // create context with no upfront defaultValue
 // without having to do undefined check all the time
-function createCtx<A>(): CreateCtx<A> {
-  const ctx = React.createContext<A | undefined>(undefined);
+function createCtx<A>(defaultContext: A): CreateCtx<A> {
+  const ctx = React.createContext<A>(defaultContext);
 
-  function useCtx(): A {
-    const c = React.useContext(ctx);
-
-    if (!c) throw new Error('useCtx must be inside a Provider with a value');
-
-    return c;
-  }
+  const useCtx = (): A => React.useContext(ctx);
```

Comment:
This is very clever~! However, I am worried that many devs will be curious that `useTheme` works without a `Provider` because usually context wasn't provided like that 🤔 

I hope there is more thing related to this to read about.

Reply:
@hyochan  Ok. I will write about it soon.